plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt

```diff
@@ -187,6 +191,24 @@ internal class LocationPuckManager(
       *targets,
       options = options
     )
+    updateMaxPulsingRadiusToFollowAccuracyRing(*radius)
+  }
+
+  /**
+   * Update maximum radius for pulsing puck to follow location accuracy radius.
+   * since pulsing radius is in pixels and location accuracy is in meters, we convert meters to pixel using
+   * projection delegate.
+   */
+  @VisibleForTesting(otherwise = PRIVATE)
+  internal fun updateMaxPulsingRadiusToFollowAccuracyRing(vararg radius: Double) {
+    if (settings.pulsingMaxRadius.toInt() == LocationComponentConstants.PULSING_MAX_RADIUS_FOLLOW_ACCURACY.toInt()) {
+      val metersPerPixelAtLocation =
+        delegateProvider.mapProjectionDelegate.getMetersPerPixelAtLatitude(
+          delegateProvider.mapCameraManagerDelegate.cameraState.center.latitude(),
+          delegateProvider.mapCameraManagerDelegate.cameraState.zoom
+        )
+      animationManager.updatePulsingRadius(radius.last() / metersPerPixelAtLocation, settings)
```

Comment:
how about passing inside this function only last radius?

Reply:
you mean not to pass `settings`? we don't have location settings reference in [PuckAnimatorManager](https://github.com/mapbox/mapbox-maps-android/blob/2a667467ccc9464456e90d41c09a05e25f0fabd0/plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/animators/PuckAnimatorManager.kt), and setting is used there to validate if pulsing is enabled.

Reply:
I mean `updateMaxPulsingRadiusToFollowAccuracyRing(vararg radius: Double)` => `updateMaxPulsingRadiusToFollowAccuracyRing(lastRadius: Double)` or just `radius`